profile-widget/profilewidget2.cpp

```diff
@@ -2044,70 +2045,75 @@ void ProfileWidget2::clearPictures()
 	pictures.clear();
 }
 
-void ProfileWidget2::updatePictures(const QModelIndex &from, const QModelIndex &to)
+void ProfileWidget2::updateThumbnail(QString filename, QImage thumbnail)
 {
-	DivePictureModel *m = DivePictureModel::instance();
-	for (int picNr = from.row(); picNr <= to.row(); ++picNr) {
-		int picItemNr = picNr - m->rowDDStart;
-		if (picItemNr < 0 || (size_t)picItemNr >= pictures.size())
-			return;
-		if (!pictures[picItemNr])
-			return;
-
-		pictures[picItemNr]->setPixmap(m->index(picNr, 0).data(Qt::UserRole).value<QPixmap>());
+	auto it = std::find_if(pictures.begin(), pictures.end(), [&filename](const PictureEntry &e)
+			       { return e.filename == filename; });
+	if (it != pictures.end() && it->thumbnail) {
+		int size = Thumbnailer::defaultThumbnailSize();
+		it->thumbnail->setPixmap(QPixmap::fromImage(thumbnail.scaled(size, size, Qt::KeepAspectRatio)));
```

Comment:
...and if we find a thumbnail, we, err, replace it? Why?
This code could really benefit from some comments...

Reply:
This is the "slot" which is asynchronously called by the thumbnailer when a thumbnail was calculated / fetched from disk. The situation is the following:

1) User switches dive.
2) For all images placeholders are shown and UI becomes responsive again.
3) Thumbnailer starts fetching / calculating thumbnails and pushes them one-by-one to the profile and the dive-photo tab.

This made switching dives distinctly snappier (already in previous release). Analogous code is in dive-photo tab, but it is now repeated here in the profile, since this PR is about disconnecting dive-photo tab and profile.

I will add a comment to the function header.

To be honest, I don't know why I used `find_if` here - it could be done by a simple loop just as well. I guess this is simply a habit of thinking more "declarative" than "imperative". No problem with changing to a loop, if you prefer.

Reply:
i think the purpose of this method is to update the thumbnail of an image, but what happens if it doesn't have one.
is there are separate method to handle the `!it->thumbnail` case? i might have missed it.

Reply:
> i think the purpose of this method is to update the thumbnail of an image, but what happens if it doesn't have one.
is there are separate method to handle the !it->thumbnail case? i might have missed it.

In the case `!it->thumbnail`, the thumbnail is not on the profile (timestamp outside of dive), therefore there is no point in updating it.